### ISSUE TITLE
JIT: a number of OSR stress tests failing with an assert

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/runtime#59785]
See https://dev.azure.com/dnceng/public/_build/results?buildId=1387134&view=results

Eg, on x64 windows
```
set COMPlus_TieredCompilation=1
set COMPlus_TC_OnStackReplacement=1
set COMPlus_TC_QuickJitForLoops=1
set COMPlus_TC_OnStackReplacement_InitialCounter=1
set COMPlus_OSR_HitLimit=1

    JIT\Regression\JitBlue\GitHub_15718\GitHub_15718\GitHub_15718.cmd [FAIL]
      
      Assert failure(PID 3084 [0x00000c0c], Thread: 2512 [0x09d0]): Assertion failed 'emitTypeSizes[TypeGet(type)] > 0' in 'System.Collections.Generic.Dictionary`2[__Canon,Nullable`1][System.__Canon,System.Nullable`1[System.Boolean]]:TryInsert(System.__Canon,System.Nullable`1[Boolean],ubyte):bool:this' during 'Generate code' (IL size 739)
      
          File: D:\workspace\_work\1\s\src\coreclr\jit\emit.h Line: 2498
          Image: C:\h\w\B39B09AE\p\corerun.exe
```

OSR stress is an OSR mode that aggressively transitions when patchpoints are reached, to try and increase coverage of OSR.

**Comment by AndyAyersMS** [comment]
In the above there's a `Nullable<bool>` arg passed in a register, and we need its size for OSR, and ask the wrong question.

**Comment by jkoritzinsky** [comment]
Fixed by #59789
